src/compiler/checker.ts

```diff
@@ -8672,6 +8672,16 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             const meaning = getMeaningOfEntityNameReference(node);
             const sym = resolveEntityName(leftmost, meaning, /*ignoreErrors*/ true, /*dontResolveAlias*/ true);
             if (sym) {
+                // If a parameter is resolvable in the current context it is also visible, so no need to go to symbol accesibility
+                if (
+                    sym.flags & SymbolFlags.FunctionScopedVariable
+                    && sym.valueDeclaration
+                ) {
+                    const parameter = sym.valueDeclaration && isParameterDeclaration(sym.valueDeclaration);

```

Comment:
The problem is that this will not work for binding elements. The original check was mean to walk up the binding elements. I this i lost something when I copied it over.

Reply:
The confusingly-named `isParameterDeclaration` does not do what you think  it does - it _does_ walk binding elements.

Reply:
Yeah, we should really rename that to `isPartOfParameterDeclaration`. See also #52283